pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -234,7 +234,7 @@ func (ku *Upgrader) upgradeAgentPools() error {
 
 			if vmProvisioningState != string(api.Succeeded) {
 				ku.logger.Infof("Deleting agent VM %s in provisioning state %s\n", *vm.Name, vmProvisioningState)
```

Comment:
do we have this log? I'm wondering if the failure cases are caused because of the node just not being in K8s at which point it should be if empty don't drain, else do. Thoughts?

Reply:
If the VM's provisioning state is not succeeded, then it really doesn't matter whether the node in K8S or not. K8S is installed as part of CSE. If that failed, then the node is not healthy from Kubernetes perspective, therefore no need to drain it. If the provisioning has failed before CSE, then K8S is not installed at all, hence again, no need to drain.

Reply:
There are other reasons a node might be unhealthy in k8s. Some of those cases there might be work loads running on the node. I agree in the case cse fails then we should just skip draining. But in that case I'm guessing it just won't have a provision state/k8s won't know it exists since the node will never report in.

Reply:
Sure, the node might become unhealthy (from K8S perspective), but the provisioning state will still be 'Succeeded' (from ARM perspective), right? The provisioning state doesn't change once the VM has been provisioned.

Reply:
ah this is the vm in arm perspective. Sorry I'm getting confused. Yeah then this should make sense.